### PR TITLE
Commit Hash in Docker Image

### PR DESCRIPTION
**Pull request by stanbrub**
- Added the _deephaven-core_ SHA the `build-ci.yml` and `edge-ci.yml` workflows
- Used COMMIT_HASH in _server_ and _server-slim_ contexts and hcls
- Tested Branch CI on my fork.  Commit hash appears in manifest:
  - ex `docker container inspect ddb093257e90` shows output that contains
  - `"org.opencontainers.image.revision": "71646c35b2987e83cf34b53b632498b8aba21332"`
- Only _server_ and _server-slim_ were changed.  Do we need any more?
- Release CI was not touched (I did come up with a possible way.)
  - ex. `curl https://api.github.com/repos/deephaven/deephaven-core/commits/v0.35.0 | grep '"sha":' | head -1 | cut -d'"' -f4`

### CODE REVIEW

File: server.hcl [hunk 1]
```diff
@@ -47,6 +47,10 @@ variable "OPENJDK_VERSION" {
     default = "21"
 }
 
+variable "COMMIT_HASH" {
+    default = "not-set"
+}
```
[the remark this answers is not in the document]
Comment (stanbrub) [reply server.hcl:52]: We could use `curl https://api.github.com/repos/deephaven/deephaven-core/commits/v0.35.0 | grep '"sha":' | head -1 | cut -d'"' -f4` to grab the commit from the release and pass it into the image bake.  (A release-ci workflow triggered by "main" would not have it set.)

File: .github/workflows/edge-ci.yml [hunk 2]
```diff
@@ -32,7 +34,10 @@ jobs:
           java-version: '11'
 
       - name: Set JAVA_HOME
-        run: echo "JAVA_HOME=${{ steps.setup-java-11.outputs.path }}" >> $GITHUB_ENV
+        id: set-java-home
+        run: |
+          echo "JAVA_HOME=${{ steps.setup-java-11.outputs.path }}" >> $GITHUB_ENV
+          echo "COMMIT_HASH="$(git rev-parse HEAD) >> $GITHUB_OUTPUT
```
Comment (devinrsmith) [inline .github/workflows/edge-ci.yml:40]: We need to do the same sort of change here to separate job, variable name, etc.

File: contexts/server-slim/Dockerfile [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ARG OPENJDK_VERSION
 ARG UBUNTU_VERSION
 ARG DEEPHAVEN_SOURCES
+ARG COMMIT_HASH
```
Comment (devinrsmith) [inline contexts/server-slim/Dockerfile:6]: Don't think we need this globally, only for relevant stages.

File: server-slim.hcl [hunk 4]
```diff
@@ -32,6 +32,10 @@ variable "OPENJDK_VERSION" {
     default = "21"
 }
 
+variable "COMMIT_HASH" {
```
Comment (devinrsmith) [inline server-slim.hcl:35]: We should change this to GIT_REVISION as well.

**Review by devinrsmith — CHANGES_REQUESTED** [review]
I think to move this label should only apply to edge / branch builds (using the Dockerfile split staging construct mentioned earlier). Also, does everything work if GIT_REVISION is set to an empty string instead of `"not-set"`?

**Comment by devinrsmith** [comment]
Also, for the release, we probably technically want to point to the tag revision and not the commit revision, if we _were_ to add the LABEL for the release. I could probably argue we should use the tag name (as opposed to the revision SHA) for the GIT_REVISION for release images, but regardless, we would want to do that separately from this PR.

```
$ git rev-parse v0.35.0     
33d520aef42bc92d22bd5190511dbc6b67f87043

$ git rev-parse release/v0.35.0
d7a2af575e3dfbbc2c429c546c66e783c93358a5
```

Note the difference.

File: server.hcl (hunk 1 above)
[the remark this answers is not in the document]
Comment (stanbrub) [reply server.hcl]: I reworked it to do deephaven-bits-released and deephaven-bits-custom.  The branch-ci and edge-ci work on my fork.  Git revision shows up for server and server-slim.  For example, with `docker container inspect <container-id> | grep revision`, you get something like `"org.opencontainers.image.revision": "b5cae98c2f11b032cdd1b9c248dc5b4a0f95314a"`.  If the GIT_REVISION ARG is not pass in, you get `"org.opencontainers.image.revision": ""`

**Review by devinrsmith — APPROVED** [review]
LGTM. Please write a merge message describing this a bit more succinctly.

**Pull request merged**